### PR TITLE
Switch from CircleCI to GHA

### PR DESCRIPTION
**Pull request by FurqanHabibi**
- Updated supported python versions:
   ["3.8", "3.9", "3.10", "3.11"]
- Runs pytest and mypy on all supported python versions
- Added yaml, eof, trailing-whitespace pre-commit
- Runs pre-commit on min supported python version

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+name: aapns workflow
+
+on: push
```
Comment (ojii) [inline .github/workflows/workflow.yml]: should also include pr

File: tests/functional/conftest.py [hunk 2]
```diff
@@ -72,19 +73,19 @@ async def server_factory(flavour):
             await server.wait()
 
 
-@pytest.fixture
+@pytest_asyncio.fixture
```
Comment (ojii) [inline tests/functional/conftest.py]: why is this needed now?

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (FurqanHabibi) [reply .github/workflows/workflow.yml]: Added PR

File: tests/functional/conftest.py (hunk 2 above)
Comment (FurqanHabibi) [reply tests/functional/conftest.py]: Removed

**Comment by jgaul3** [comment]
@FurqanHabibi @ojii would be nice to have these changes in the latest version. Changes look good as far as I can see

Also, possible to re-run the tests? Seems like they may have resulted in `Event loop is closed` errors...

**Comment by ojii** [comment]
> @FurqanHabibi @ojii would be nice to have these changes in the latest version. Changes look good as far as I can see
> 
> Also, possible to re-run the tests? Seems like they may have resulted in `Event loop is closed` errors...

I think tests are fubar, I'm trying to fix them (over in #42, struggling with golang right now)
